c10/xpu/XPUException.h

```diff
@@ -0,0 +1,19 @@
+#pragma once
+
+#include <c10/util/Exception.h>
+#include <sycl/sycl.hpp>
+
+namespace c10::xpu {
+
+static sycl::async_handler asyncHandler = [](sycl::exception_list el) {
+  for (auto& e : el) {
+    try {
+      std::rethrow_exception(e);
+    } catch (sycl::exception& e) {
+      TORCH_WARN("SYCL Exception: ", e.what());
+      throw;
+    }
+  }
+};
```

Comment:
Correct me if I'm wrong, but static global variable will force compiler to create a separate instance every time header is included, even if variable is not referenced later, while `static inline` function will be materialized multiple times, but will be thrown away if not used.

Also, not much reason to iterate over the list of exceptions as one is going to throw first one anyway
```suggestion
static inline asyncHandle(sycl::exception_list el) {
  if (el.empty()) {
    return;
  }
  try {
    std::rethrow_exception(el.front());
  } catch (sycl::exception& e) {
    TORCH_WARN("SYCL Exception: ", e.what());
    throw;
  }
}
```

Reply:
Done. `sycl::exception_list` has no member `empty` and `front()`, use `size` and `begin` instead.